op-challenger/game/fault/contracts/oracle.go

```diff
@@ -37,3 +42,35 @@ func (c PreimageOracleContract) AddGlobalDataTx(data *types.PreimageOracleData)
 	call := c.contract.Call(methodLoadKeccak256PreimagePart, new(big.Int).SetUint64(uint64(data.OracleOffset)), data.GetPreimageWithoutSize())
 	return call.ToTxCandidate()
 }
+
+func (c PreimageOracleContract) InitLPP(ctx context.Context, uuid *big.Int, partOffset uint32, claimedSize uint32) (txmgr.TxCandidate, error) {
+	call := c.contract.Call(methodInitLPP, uuid, partOffset, claimedSize)
+	return call.ToTxCandidate()
+}
+
+func (c PreimageOracleContract) AddLeavesLPP(ctx context.Context, uuid *big.Int, input []byte, stateCommitments [][32]byte, finalize bool) (txmgr.TxCandidate, error) {
+	call := c.contract.Call(methodAddLeavesLPP, uuid, input, stateCommitments, finalize)
+	return call.ToTxCandidate()
+}
+
+// LibKeccakStateMatrix is an auto generated low-level Go binding around an user-defined struct.
+type LibKeccakStateMatrix struct {
+	State [25]uint64
+}
+
+// PreimageOracleLeaf is an auto generated low-level Go binding around an user-defined struct.
+type PreimageOracleLeaf struct {
+	Input           []byte
+	Index           *big.Int
+	StateCommitment [32]byte
+}
+
+func (c PreimageOracleContract) ChallengeLPP(ctx context.Context, claimant common.Address, uuid *big.Int, stateMatrix LibKeccakStateMatrix, preState PreimageOracleLeaf, preStateProof [][32]byte, postState PreimageOracleLeaf, postStateProof [][32]byte) (txmgr.TxCandidate, error) {
+	call := c.contract.Call(methodChallengeLPP, claimant, uuid, stateMatrix, preState, preStateProof, postState, postStateProof)
+	return call.ToTxCandidate()
+}
+
+func (c PreimageOracleContract) SqueezeLPP(ctx context.Context, claimant common.Address, uuid *big.Int, stateMatrix LibKeccakStateMatrix, preState PreimageOracleLeaf, preStateProof [][32]byte, postState PreimageOracleLeaf, postStateProof [][32]byte) (txmgr.TxCandidate, error) {
```

Comment:
This feels pretty low level too. The state matrix probably should be the `KeccakState` from https://github.com/ethereum-optimism/optimism/pull/9015 and then convert it to the contract form internally.  Not exactly sure what the right structure for the merkle proofs are - that may be what we get out of geth's trie implementation.

So would be good to use our StateMatrix implementation but I'd be ok with leaving the proofs as-is for now as long as we do adjust them to match what we get from the geth trie when we actually have that implemented.